graph/assemblygraph.h

```diff
@@ -191,10 +198,60 @@ class AssemblyGraph : public QObject
     bool attemptToLoadSequencesFromFasta();
     long long getTotalLengthOrphanedNodes() const;
     bool useLinearLayout() const;
+
+    // FIXME: This belongs to UI controller!
     bool nodeHasBlastHit(const DeBruijnNode *node) const;
     bool nodeOrReverseComplementHasBlastHit(const DeBruijnNode *node) const;
     const std::vector<std::shared_ptr<BlastHit>> &getBlastHits(const DeBruijnNode *node) const;
 
+    bool hasCustomColour(const DeBruijnNode* node) const {
+        auto it = m_nodeColors.find(node);
+        return it != m_nodeColors.end() && it->second.isValid();
+    }
+    QColor getCustomColour(const DeBruijnNode* node) const {
+        auto it = m_nodeColors.find(node);
+        return it == m_nodeColors.end() ? QColor() : it->second;
+    }
+    void setCustomColour(const DeBruijnNode* node, QColor color) {
+        m_nodeColors[node] = color;
+    }
+    QString getCustomLabel(const DeBruijnNode* node) const {
```

Comment:
I also was needed in such boilerplate and made a function:
```cpp
template <typename K, typename V>
const V &getFromMapOrDefaultConstructed(const phmap::parallel_flat_hash_map<K, V> &map, const K &key) {
    auto it = map.find(key);
    if (it != map.end()) {
        return it->second;
    } else {
        static const V defaultConstructed{};
        return defaultConstructed;
    }
}
```

So there aren't unwanted copies and empty object creations.

Reply:
Fair enough. So, let's switch when it lands in your PR